credentialMixin.js

```diff
@@ -79,31 +81,53 @@ export const credentialMixin = {
       return this.issuer.name;
     },
     credentialImage() {
+      if(!this.credential) {
+        return '';
+      }
       if(this.credential.image) {
         return this.credential.image;
       }
-      if(this.issuer.image) {
-        return this.issuer.image;
-      }
-      if(this.issuer.logo) {
-        return this.issuer.logo;
+      const {credentialSubject = {}} = this.credential;
+      if(credentialSubject.image) {
+        return credentialSubject.image;
       }
       return '';
     },
     truncateDescription() {
+      // if there is no description don't truncate it
+      if(!(this.credential && this.credential.description)) {
+        return false;
+      }
       const {length} = this.credential.description;
       return length > MAX_CHARACTER_COUNT;
+    },
+    /**
+      * Takes all of the types after the initial VC one
+      * and turns them into a string.
+    */
+    credentialType() {
+      if(!this.credential) {
+        return '';
+      }
+      return this.credential.type
+        .filter(type => type !== 'VerifiableCredential')
+        .join(', ');
```

Comment:
This is weird. If we're going to surface types, shouldn't we be expanding them? They need to be globally unambiguous. That gets into a mess. We should be using a `name` as done before, IMO.

Reply:
Longely, this PR has been in PR hell for to long. I just removed everything not related to the bug being fixed i.e. with truncate description. So marking this as resolved.